x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Literal.java

```diff
@@ -122,7 +141,20 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        String str = String.valueOf(value);
+        String str;
+        if (dataType == KEYWORD || dataType == TEXT) {
+            str = BytesRefs.toString(value);
+        } else if (dataType == VERSION && value instanceof BytesRef br) {
+            str = new Version(br).toString();
```

Comment:
Is this an improvement? Or were Versions a readable String before?

Reply:
We had some cases (mostly unit tests) where we passed plain strings to VERSION literals.
Even for tests it's not a good thing, as we are testing something that is different than the actual production execution.